### PR TITLE
json_util.pyi: add some stubs

### PR DESCRIPTION
**Pull request by gomonuk**

### CODE REVIEW

File: typings/bson/json_util.pyi [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from typing import Any
 
 def loads(s: str) -> Any: ...
-def dumps(obj: object) -> str: ...
+def dumps(obj: object, *args, **kwargs) -> str: ...
+def default(obj: object, json_options: Any = ...) -> dict: ...
```
Comment (sbdchd) [inline typings/bson/json_util.pyi:5]:
```suggestion
def default(obj: object, json_options: Any = ...) -> Any: ...
```

Looking at the source code for  `default`, it sometimes returns `SON`, `dict[str, Any]`, `bytes` depending on the param

we could add a bunch of `overload`s, but fine to type it as returning `Any` for now

https://github.com/mongodb/mongo-python-driver/blob/acfa7b615c5d263a2fc94d7746771beb6fff4016/bson/json_util.py#L766-L859

File: typings/bson/json_util.pyi [hunk 2]
```diff
@@ -1,4 +1,5 @@
 from typing import Any
 
 def loads(s: str) -> Any: ...
-def dumps(obj: object) -> str: ...
+def dumps(obj: object, *args, **kwargs) -> str: ...
```
Comment (sbdchd) [inline typings/bson/json_util.pyi:4]:
```suggestion
def dumps(obj: object, *args: Any, **kwargs: Any) -> str: ...
```

File: typings/bson/json_util.pyi (hunk 1 above)
Comment (gomonuk) [reply typings/bson/json_util.pyi]: Yeah, I noticed the dicts only :))

File: typings/bson/json_util.pyi (hunk 2 above)
Comment (gomonuk) [reply typings/bson/json_util.pyi]: Does that make sense? In both cases, you can pass any variable

**Comment by sbdchd** [comment]
Looks good, thanks!

**Pull request merged**